### PR TITLE
Add automatic generation of best learner for given problem

### PR DESCRIPTION
**Pull request by jakob-r**
Please discuss here: Issue #80

### CODE REVIEW

**Comment by berndbischl** [comment]
I really dislike this. 

1) the nugget stuff should go in mlr, like i said in the issue.

2) the basewrapper construction is really "weird" and overly complex

3) having an "auto construct" function for certain problems although i like.
but then go the whole way and make it possible to return also the random forest for mixed problems

**Comment by berndbischl** [comment]
we aggreed that 

a) jakob does the mlr PR with param name "nugget.stability"
b) we have a method makeMBODefaultLearner or whatever

**Comment by jakob-r** [comment]
Okay - I changed it so that it doesn't use a wrapper but relies on the [mlr PR](https://github.com/mlr-org/mlr/pull/885).

File: R/makeMboLearner.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    } else if (is.null(getHyperPars(base.learner)$nugget) && is.null(getHyperPars(base.learner)$nugget.stability)) {
+      learner = setHyperPars(base.learner, nugget.stability = 10^-8)
+    } else {
+      learner = setHyperPars(base.learner, par.vals)
+    }
+  } else {
+    learner = makeLearner("regr.randomForest", predict.type = "se", par.vals = par.vals)
+  }
+  return(learner)
+}
```
Comment (jakobbossek) [inline R/makeMboLearner.R]: Personally, I prefer:
if (c1)
  return(r1)
if (c2)
  return(r2)
...
return(zn)

**Comment by jakobbossek** [comment]
Looks good to me 👍

**Comment by berndbischl** [comment]
Ok, merge this asap.

but note the conflicts with PR #255

**Comment by berndbischl** [comment]
just do the changes for PR #256 here, merge this, then close both things, pls
now

File: R/makeMboLearner.R (hunk 1 above)
Comment (jakob-r) [inline R/makeMboLearner.R]: Proably outdated now. Is this style okay?

**Comment by jakob-r** [comment]
@berndbischl Combined the stuff. Can you have a look?

**Comment by jakob-r** [comment]
@berndbischl Improved description part. Can you have a look?

**Comment by jakobbossek** [comment]
I have nothing to complain. Looks fine.
IMHO we can merge.
@berndbischl: what is you opinion?

**Comment by jakobbossek** [comment]
Just looked at this. Looks good. Merging.

**Pull request merged**